pddlgym/core.py

```diff
@@ -78,6 +79,36 @@ def get_successor_state(state, action, domain, raise_error_on_invalid_action=Fal
 
     return state
 
+
+def get_successor_states(state, action, domain, raise_error_on_invalid_action=False,
```

Comment:
Together with the method `get_all_possible_transitions(self, action)` that was suggested, I found it valuable to add this function that gets the successor states from a passed state and action, instead of getting the state from the current env object.
`get_all_possible_transitions` then uses this by passing its current state.

Reply:
It looks like this method is the same as get_successor_state except that get_all_transitions=True in the call to _apply_effects. It might be better to just add an argument to get_successor_state, called get_all_possible or something, that is False by default? And then just pass that argument through to _apply_effects. I think this might help keep the code cleaner because then we don't need this new get_successor_states method. What do you think?

Reply:
That's true, I agree. My only concern would be minor naming issues (`get_successor_states` means more that possibly more states can be return instead of just one). What would ou think about keeping get_successor_states but only to call the new get_sucessor_state function with the new flag marked as True?

Reply:
Yeah, that sounds good!

Reply:
Awesome!